probe-rs/src/architecture/arm/core/armv7a.rs

```diff
@@ -1110,8 +1213,49 @@ impl MemoryInterface for Armv7a<'_> {
 
     fn read_32(&mut self, address: u64, data: &mut [u32]) -> Result<(), Error> {
         self.halted_access(|core| {
-            for (i, word) in data.iter_mut().enumerate() {
-                *word = core.read_word_32(address + ((i as u64) * 4))?;
+            let count = data.len();
+            if count > 2 {
+                // Save r0
+                core.prepare_r0_for_clobber()?;
+                core.set_r0(valid_32bit_address(address)?)?;
+
+                let mut banked = core.banked_access()?;
+
+                // Ignore any errors encountered here -- they will set a Data Abort
+                // which we will pick up in `check_and_clear_data_abort()`
+                if banked
+                    .with_dcc_fast_mode(|banked| {
+                        // LDC p14, c5, [r0], #4
+                        banked.set_itr(build_ldc(14, 5, 0, 4))?;
+
+                        // Throw away the first value, which is from a previous operation
+                        let _ = banked.dtrrx()?;
+
+                        // Continually write the tx register, which will auto-increment.
+                        // Don't load the final value -- we'll read it after we're done.
```

Comment:
Why do we read the last value after we are done? It must be something in the greater scheme that I am missing.

Reply:
Because the result that pops out on the Rx side is delayed by one cycle, and is the result from the previous iteration. So each time you read while in DCC Fast mode it reads the value from the subsequent address.

We need to drop out of DCC Fast mode before reading the last value, otherwise it will get triggered again and will end up reading past the buffer. If the address isn't valid then that could generate an abort.

Reply:
I've updated it to describe this in more detail.